iaso/tests/api/test_tasks.py

```diff
@@ -1,26 +1,127 @@
-from beanstalk_worker.services import TestTaskService
-from iaso.test import APITestCase
-from ...models import OrgUnit, OrgUnitType, Account, Project, SourceVersion, DataSource, Task
+import datetime
+import time_machine
+
+from django.utils import timezone
+
+from iaso import models as m
+from iaso.api.tasks import TaskSerializer
+from iaso.test import APITestCase, TestCase
+
+
+DT = datetime.datetime(2023, 12, 20, 15, 0, 0, 0, tzinfo=timezone.utc)
+
+
+@time_machine.travel(DT, tick=False)
+class TaskSerializerSerializerTestCase(TestCase):
+    """
+    Test Task serializer.
+    """
+
+    @classmethod
+    def setUpTestData(cls):
+        data_source = m.DataSource.objects.create(name="Data source")
+        version = m.SourceVersion.objects.create(number=1, data_source=data_source)
+        account = m.Account.objects.create(name="Account", default_version=version)
+        user = cls.create_user_with_profile(
+            username="user", account=account, permissions=["iaso_sources", "iaso_data_tasks"]
+        )
+
+        cls.task = m.Task.objects.create(
+            progress_value=1,
+            end_value=1,
+            account=user.iaso_profile.account,
+            launcher=user,
+            status="Success",
+            name="org_unit_bulk_update",
+            result="324.49 sec, processed 5262 org units",
+            params={},
+        )
+
+        cls.task_polio = m.Task.objects.create(
+            progress_value=1,
+            end_value=1,
+            account=user.iaso_profile.account,
+            launcher=user,
+            status="Success",
+            name="create_polio_notifications_async",
+            result="3768 polio notifications created.",
+            params={
+                "args": [],
+                "kwargs": {"pk": 10},
+                "method": "create_polio_notifications_async",
+                "module": "plugins.polio.models",
+            },
+        )
+
+    def test_serialize_instance_of_task(self):
+        serializer = TaskSerializer(self.task)
+        self.assertEqual(
+            serializer.data,
+            {
+                "id": self.task.pk,
+                "created_at": DT.timestamp(),
+                "started_at": None,
+                "ended_at": None,
+                "progress_value": 1,
+                "end_value": 1,
+                "launcher": {
+                    "first_name": "",
+                    "last_name": "",
+                    "username": "user",
+                },
+                "result": "324.49 sec, processed 5262 org units",
```

Comment:
I may be wrong, but I think that this value should gotten from self.task like self.task.pk

Reply:
I don't understand your comment.
`result` is stored as text in `Task.result` and defined just above in `setUpTestData()`.

Reply:
I was just wondering if instead of writing "324.49 sec, processed 5262 org units" we can write self.task.result so that when its value change it doesn't break the test. but it's not blocking

Reply:
Ah ok, understood. Nope, in fact I want the test to break explicitly if the value change!